src/PythonMigrationViewExtension/GraphPythonDependencies.cs

```diff
@@ -18,10 +23,72 @@ internal GraphPythonDependencies(ViewLoadedParams viewLoadedParams)
         internal bool ContainsIronPythonDependencies()
         {
             var workspace = ViewLoaded.CurrentWorkspaceModel;
+
             if (workspace == null)
                 throw new ArgumentNullException(nameof(workspace));
 
-            return workspace.Nodes.Any(n => IsIronPythonNode(n));
+            if (workspace.Nodes.Any(n => IsIronPythonNode(n)))
+            {
+                return true;
+            }
+
+            // Check if any of the custom nodes has IronPython dependencies in it. 
+            var customNodeManager = ViewLoaded.StartupParams.CustomNodeManager;
+            var customNodes = workspace.Nodes.OfType<Function>();
+
+            return CustomNodeContainsIronPythonDependency(customNodes, customNodeManager);           
+        }
+
+        internal bool CustomNodeContainsIronPythonDependency(IEnumerable<Function> customNodes, ICustomNodeManager customNodeManager)
```

Comment:
I think it would be good to write /// summary for this function.

Reply:
to help clarify my understanding - would a good summary be: 

```
This function traverses all custom nodes and their subtrees and marks them as containing ironPython dependencies if any of their child nodes contain IronPython.

```

Reply:
in addition to a summary, I think this function should have a unit test written if possible.

Reply:
That would be a good summary but with condition that it traverses until the first CN with an IronPython dependency is found. Also had to make the method static, so as to call it from the tests. Added a unit test for this function as well.